docs/development/doc_guide.rst

```diff
@@ -1,76 +1,633 @@
+      -------
+      difference : float
+          The difference between ``a`` and ``b``.
+
+      Raises
+      ------
+      `ValueError`
+          If ``a`` or ``b`` is `~numpy.inf`.
+
+      Warns
+      -----
+      `UserWarning`
+          If ``a`` or ``b`` is `~numpy.nan`.
+
+      See Also
+      --------
+      add : Add two numbers.
+
+      Notes
+      -----
+      The "Notes" section provides extra information that cannot fit in
+      the extended summary near the beginning of the docstring. This
+      section should include a discussion of the physics behind a
+      particular concept that should be understandable to someone who is
+      taking their first plasma physics class. This section can
+      include a derivation of the quantity being calculated or a
+      description of a particular algorithm.
+
+      The next section contains example references to a journal article
+      [1]_ and a book [2]_.
+
+      References
+      ----------
+      .. [1] J. E. Foster, `Plasma-based water purification: Challenges and
+         prospects for the future <https://doi.org/10.1063/1.4977921>`_,
+         Physics of Plasmas, 22, 05501 (2017).
+
+      .. [2] E. Gamma, R. Helm, R. Johnson, J. Vlissides, `Design Patterns:
+         Elements of Reusable Object-Oriented Software
+         <https://www.oreilly.com/library/view/design-patterns-elements/0201633612/>`_
+
+      Examples
+      --------
+      Include a few example usages of the function here. Start with simple
+      examples and then increase complexity when necessary.
+
+      >>> from package.subpackage.module import subtract
+      >>> subtract(9, 6)
+      3
+
+      Here is an example of a multi-line function call.
+
+      >>> subtract(
+      ...     9, 6, switch_order=True,
+      ... )
+      -3
+
+      PlasmaPy's test suite will check that these commands provide the
+      output that follows each function call.
+      """
+      if np.isinf(a) or np.isinf(b):
+          raise ValueError("Cannot perform substraction operations involving infinity.")
+
+      warnings.warn("The subtract function encountered a nan value.", UserWarning)
+
+      return b - a if switch_order else a - b
+
+Template docstring
+~~~~~~~~~~~~~~~~~~
+
+This template docstring may be copied into new functions. Usually only
+some of the sections will be necessary for a particular function, but
+any sections that are included should be in the order provided.
+
+.. code-block:: python
+  :caption: Docstring template
+
+  def sample_function():
+      r"""
+      Return ...
+
+      Parameters
+      ----------
+
+      Returns
+      -------
+
+      Raises
+      ------
+
+      Warns
+      -----
+
+      See Also
+      --------
+
+      Notes
+      -----
+
+      References
+      ----------
+
+      Examples
+      --------
+
+      """
+
+
+Documentation guidelines
+========================
+
+This section contains guidelines and best practices for writing
+PlasmaPy documentation.
```

Comment:
Add a bullet stating that if the main docstring should only be 2-4 sentences followed by the "Parameters" section.  If more detail needs to be added, then the "Notes" section should be used to expand on the main component of the docstring.